Modules/_ctypes/cfield.c

```diff
@@ -1087,6 +1091,34 @@ d_get(void *ptr, Py_ssize_t size)
     return PyFloat_FromDouble(val);
 }
 
+#ifdef __STDC_IEC_559_COMPLEX__
+#if defined(__clang__) && __has_builtin(__builtin_complex) && \
+    !defined(CMPLX)
+#  define CMPLX(x, y) __builtin_complex ((double) (x), (double) (y))
+#endif
+static PyObject *
+C_set(void *ptr, PyObject *value, Py_ssize_t size)
+{
+    Py_complex c = PyComplex_AsCComplex(value);
+    double complex x = CMPLX(c.real, c.imag);
+
+    if (c.real == -1 && PyErr_Occurred()) {
```

Comment:
Don't you need this check for `Py_complex c = ...` above?

Reply:
No, both components of Py_complex are defined in case of error.

Reply:
Actually, it was just to avoid creating `x` (not because we access `c.imag`) but I'm fine with your code too!